docs/src/main/tut/best-practices.md

```diff
@@ -0,0 +1,322 @@
+---
+layout: docs
+title: Best-Practices
+position: 3
+---
+
+## Best Practices
+
+* [The big picture](#the-big-picture)
+* [Picking a JSON library](#picking-a-json-library)
+* [Do not block an endpoint](#do-not-block-an-endpoint)
+* [Use TwitterServer](#use-twitterserver)
+* [Monitor your application](#monitor-your-application)
+* [Picking HTTP statuses for responses](#picking-http-statuses-for-responses)
+* [Configuring Finagle](#configuring-finagle)
+* [Finagle Filters vs. Finch Endpoints](#finagle-filters-vs-finch-endpoints)
+* [Pulling it all Together](#pulling-it-all-together)
+
+
+### The big picture
+
+There were a lot of discussions and thoughts on "What does an idiomatic Finch program look like?"
+(see the [Best Practices and Abstractions][issue263] issue and the [Future Finch][future-finch]
+writeup), but we're not sure we should end up sticking with a "one-size-fits-all" style in Finch.
+In fact, zero lines of Finch's code were written with some particular style in mind, rather
+reasonable composition and reuse of its building blocks. That said, Finch will always be a
+_library_ that doesn't promote any concrete style for organizing a code base (frameworks usually
+do that), but does promote composability and compile-time guarantees for its abstractions.
+
+Given that major Finch's abstractions are pretty generic, you can make them
+[have any faces you like][faceless] as long as it makes programming fun again: some of the Finch
+users write [Jersey-style][diar] programs; some of them stick with [CRUD-style][issue263] ones;
+others use macros to group endpoints in [controllers-like][finchrich] structures.
+
+Note that all of the Finch examples are written in the "Vanilla Finch" style, where no additional
+levels of indirections are layered on top of endpoints.
+
+### Picking a JSON library
+
+It's highly recommended to use [Circe][circe], whose purely functional nature aligns very well with
+Finch (see [Cookbook docs](cookbook#circe) on how to enable Circe support in Finch). In addition to
+[compile-time derived decoders][generic-decoders] and encoders, Circe has
+[great performance][circe-performance] and very useful features such as case class patchers and
+incomplete decoders.
+
+Case class patchers are extremely useful for `PATCH` and `PUT` HTTP endpoints, when it's required to
+_update_ the case class instance with new data parsed from a JSON object. In Finch, Circe's case
+class patchers are usually represented as `Endpoint[A => A]`, which
+
+1. parses an HTTP request for a partial JSON object that contains the fields which need to be
+   updated and
+2. represents that partial JSON object as a function `A => A` that takes a case class and updates
+   it with all the values from a JSON object
+
+```tut:silent
```

Comment:
tut:silent compiles the code block but hides the scala console output if you just use ```tut you get code as if you've use the REPL. In some situations I've used it because the output was true / false but mostly I've hided it, because I didn't see the added value of it.